fabric/src/main/java/dev/architectury/registry/registries/fabric/RegistriesImpl.java

```diff
@@ -79,27 +99,50 @@
         }
         
         @Override
-        @NotNull
         public <T> RegistrarBuilder<T> builder(Class<T> type, ResourceLocation registryId) {
-            return new RegistrarBuilderWrapper<>(FabricRegistryBuilder.createSimple(type, registryId));
+            return new RegistrarBuilderWrapper<>(modId, FabricRegistryBuilder.createSimple(type, registryId));
+        }
+    }
+    
+    public static class RegistryEntryId<T> {
```

Comment:
This could just be a record, couldn't it?

Reply:
Wanted easier port back to 1.16

Reply:
Fair enough